### PR TITLE
Procedure using vespa-cli

### PR DESCRIPTION
**Pull request by kkraune**
FYI @bratseth 

This is the manual part, lets have a run for the auto testing

I will add a test assert in next PR + add some more finishing to the readme

### CODE REVIEW

File: vespa-cloud/document-processing/README.md [hunk 1]
```diff
@@ -21,115 +21,139 @@ and the asynchronous [Document API](https://docs.vespa.ai/en/document-api-guide.
...
+for Windows, Linux and macOS.
 
 
-**Clone sample apps:**
 
-<pre data-test="exec">
-$ git clone --depth 1 https://github.com/vespa-engine/sample-apps.git
-$ cd sample-apps/vespa-cloud/document-processing
-</pre>
+###Create your application in the Vespa Cloud:
```
Comment (mpolden) [inline vespa-cloud/document-processing/README.md:43]: Need a space after `###` for this to render properly.

File: vespa-cloud/document-processing/README.md [hunk 2]
```diff
@@ -21,115 +21,139 @@ and the asynchronous [Document API](https://docs.vespa.ai/en/document-api-guide.
 
 Flow:
 1. Feed album document with the _music_ schema
-1. Look up in the _lyrics_ schema if album with given ID has lyrics stored
-1. Store album with lyrics in the _music_ schema
+2. Look up in the _lyrics_ schema if album with given ID has lyrics stored
+3. Store album with lyrics in the _music_ schema
```
Comment (mpolden) [inline vespa-cloud/document-processing/README.md:25]: Using only `1.` allows the Markdown renderer to generate the sequence.

File: vespa-cloud/document-processing/README.md [hunk 3]
```diff
@@ -21,115 +21,139 @@ and the asynchronous [Document API](https://docs.vespa.ai/en/document-api-guide.
...
+Use the tenant and application name from step 2.
+
+<!-- Override VESPA_CLI_HOME to work around container filesystem limitations and set application used for automatic testing -->
+<!-- $PWD is set to $SD_DIND_SHARE_PATH by screwdriver.yaml - a special Docker-in-Docker path -->
+<pre data-test="exec" style="display:none">
+$ export VESPA_CLI_HOME=$PWD/.vespa TMPDIR=$PWD/.tmp
+$ mkdir -p $TMPDIR
+$ vespa config set target cloud
+$ vespa config set application vespa-team.document-processing.my-instance
+</pre>
```
Comment (mpolden) [inline vespa-cloud/document-processing/README.md:81]: Is there a way we can hide this? It shows up for me when displaying the rendered readme: https://github.com/vespa-engine/sample-apps/blob/5910ec99aba9aece06dc52e74be64f50736ffbb9/vespa-cloud/document-processing/README.md

File: vespa-cloud/document-processing/README.md [hunk 4]
```diff
@@ -21,115 +21,139 @@ and the asynchronous [Document API](https://docs.vespa.ai/en/document-api-guide.
...
+### Create a user API key:
+```
+$ vespa api-key
+```
+Follow the instructions from the command to register the key.
+
+<!-- Write API key used in automatic testing -->
+<!-- $VESPA_TEAM_API_KEY is a base64-encoded PEM-encoded EC private key in PKCS#8 format. -->
+<!-- However, the current version of Vespa CLI only supports PEM-encoded raw EC private keys, so we have to convert it -->
+<pre data-test="exec" style="display:none">
```
Comment (mpolden) [inline vespa-cloud/document-processing/README.md:94]: Same as above.

File: vespa-cloud/document-processing/README.md [hunk 5]
```diff
@@ -21,115 +21,139 @@ and the asynchronous [Document API](https://docs.vespa.ai/en/document-api-guide.
...
+
+<pre>
 $ cat src/test/resources/A-Head-Full-of-Dreams.json
 </pre>
 
 
-**Review logs:**
 
+### Review logs:
 Use the [console](http://console.vespa.ai) to download logs, then inspect what happened:
```
Comment (mpolden) [inline vespa-cloud/document-processing/README.md:157]: It's also possible to use `vespa log`.

File: vespa-cloud/document-processing/README.md [hunk 6]
```diff
@@ -21,115 +21,139 @@ and the asynchronous [Document API](https://docs.vespa.ai/en/document-api-guide.
...
+### Build and deploy the application:
+```
+$ mvn package vespa:deploy
+```
+The first deployment may take a few minutes.
+
+<!-- Set instance used in automatic testing -->
+<pre data-test="exec" style="display:none">
+$ mvn package vespa:deploy -Dinstance=my-instance -DapiKeyFile=$VESPA_CLI_HOME/vespa-team.api-key.pem
 </pre>
```
Comment (mpolden) [inline vespa-cloud/document-processing/README.md:118]: Same as above.

File: vespa-cloud/document-processing/README.md (hunk 2 above)
Comment (kkraune) [reply vespa-cloud/document-processing/README.md:25]: yup :-) auto-corrected by WebStorm

**Comment by kkraune** [comment]
taking a step back, I think we cannot have such guides in READMEs as we cannot hide elements needed for auto testing. maybe better to put this in a guide in cloud docs instead

**Comment by kkraune** [comment]
discussed with Jon: will move the steps to cloud site to a guide a la https://cloud.vespa.ai/en/getting-started-java and link from here instead

**Pull request closed without merging**